tools/clang/lib/Analysis/UninitializedValues.cpp

```diff
@@ -496,11 +496,10 @@ void ClassifyRefs::VisitCallExpr(CallExpr *CE) {
       if (FD->getNumParams() > ParamIdx) {
         ParmVarDecl *PD = FD->getParamDecl(ParamIdx);
         bool HasIn = PD->hasAttr<HLSLInAttr>();
-        bool HasOut = PD->hasAttr<HLSLOutAttr>();
         bool HasInOut = PD->hasAttr<HLSLInOutAttr>();
         // If we have an in annotation or no annotation (implcit in), this is a
         // use not an initialization.
```

Comment:
Is this comment stale? I don't think it describes what's happening accurately even before your change.

Reply:
Is `in` added elsewhere by default for user function params?

Reply:
No, this only covers explicit annotations. Pass-by-value parameters (including explicitly annotated `in`) should automatically be treated as uses because they’ll pretty much always have an `LValueToRValue` cast which is a use.

Thinking on it we need the `in` check here is to cover the case of `in out` vs `inout`, but otherwise we wouldn’t need to check `in` at all.